client/components/ProductDetail/ProductDetail.jsx

```diff
@@ -13,159 +13,80 @@ import ShareButtons from './ShareButtons.jsx';
-        break;
-      }
-    }
-    console.log(this.state.selectedSize);
-  }
+const ProductDetail = ({ product, productReviews, addToCart }) => {
+  console.log('>>>>>', product);
 
-  changeQuantity(quantity) {
-    this.setState({
-      selectedQuantity: quantity,
-    });
-  }
+  const [style, setStyle] = useState(product.styles.length > 0 ? product.styles[0] : null);
+  const [photo, setPhoto] = useState(
+    product.styles.length > 0 ? product.styles[0].photos[0] : null
+  );
 
-  changeStyle(style) {
-    this.setState({
-      currentStyle: style,
-      currentPhoto: style.photos[0],
-    });
-  }
+  let rating = totalRating(productReviews.ratings);
+  let allRatings = productReviews.numReviews;
 
-  changePhoto(photo) {
-    this.setState({
-      currentPhoto: photo,
-    });
-  }
-
-  componentDidUpdate(prevProps, prevState, snapshot) {
-    const { id } = this.props.product;
-    if (id && JSON.stringify(prevProps) !== JSON.stringify(this.props)) {
-      api.getProductData({ product_id: id }).then((res) => {
-        this.setState({
-          productStyles: res.styles,
-          currentProduct: res,
-          currentStyle: res.styles[0],
-          currentPhoto: res.styles[0].photos[0],
-          rating: totalRating(this.props.productReviews.ratings),
-          allRatings: this.props.productReviews.numReviews,
-        });
-      });
+  useEffect(() => {
+    if (product.styles.length > 0) {
+      setStyle(product.styles.length > 0 ? product.styles[0] : null);
+      setPhoto(product.styles.length > 0 ? product.styles[0].photos[0] : null);
+    } else {
+      setStyle(null);
+      setPhoto(null);
     }
-  }
-
-  render() {
-    const {
-      currentProduct,
-      productStyles,
-      currentStyle,
-      currentPhoto,
-      selectedSize,
-      selectedQuantity,
-      rating,
-      allRatings,
-    } = this.state;
-    return (
-      <Container id='ProductDetail'>
-        <h3>Product Details</h3>
-        {currentProduct && (
-          <ProductContainer>
-            <DisplayContainer>
-              <ImageGalleryContainer>
-                <ImageGallery currentStyle={currentStyle} changePhoto={this.changePhoto} />
-              </ImageGalleryContainer>
-              <ImageContainer>
-                <ProductImage currentPhoto={currentPhoto} />
-              </ImageContainer>
-            </DisplayContainer>
-            <ProductInfoContainer>
-              <ReviewsContainer>
-                <Stars number={rating} />
-                <ScrollToReviews allRatings={allRatings} />
-              </ReviewsContainer>
-              <br></br>
-              <div>
-                <b>Category:</b> {currentProduct.category}
-              </div>
-              <p>
-                <b>Product Name: </b>
-                {currentProduct.name}
-              </p>
+  }, [product.id]);
+
+  return (
+    <Container id='ProductDetail'>
+      <h3>Product Details</h3>
+      <ProductContainer>
+        {style && (
```

Comment:
[consider the consequences](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx)

p.s. i use `&&` all the time, just something to keep in mind.